ListLibrary/MyList.cs

```diff
@@ -0,0 +1,519 @@
+        }
+
+        public int RemoveAll(T element)
+        {
+            if (element == null)
+            {
+                throw new ArgumentNullException("Item can't be null");
+            }
+
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            int result = 0;
+
+            for (int i = 0; i < _size; i++)
+            {
+                if (_items[i].Equals(element))
+                {
+                    RemoveAt(i);
+                    result++;
+                }
+            }
+
+            return result;
+        }
+
+        #endregion
+
+        #region Search and Sort
+
+        public int IndexOf(T item)
+        {
+            if (item == null)
+            {
+                throw new ArgumentNullException("Item");
+            }
+
+            int result = -1;
+
+            for (int i = 0; i < _size; i++)
+            {
+                if (_items[i].Equals(item))
+                {
+                    result = i;
+                    break;
+                }
+            }
+
+            return result;
+        }
+
+        public T Max()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            T result = _items[0];
+
+            for (int i = 1; i < _size; i++)
+            {
+                if (_items[i].CompareTo(result) == 1)
+                {
+                    result = _items[i];
+                }
+            }
+
+            return result;
+        }
+
+        public T Min()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            T result = _items[0];
+
+            for (int i = 1; i < _size; i++)
+            {
+                if (_items[i].CompareTo(result) == -1)
+                {
+                    result = _items[i];
+                }
+            }
+
+            return result;
+        }
+
+        public int IndexOfMax()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            int result = 0;
+
+            for (int i = 1; i < _size; i++)
+            {
+                if (_items[i].CompareTo(_items[result]) == 1)
+                {
+                    result = i;
+                }
+            }
+
+            return result;
+        }
+
+        public int IndexOfMin()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            int result = 0;
+
+            for (int i = 1; i < _size; i++)
+            {
+                if (_items[i].CompareTo(_items[result]) == -1)
+                {
+                    result = i;
+                }
+            }
+
+            return result;
+        }
+
+        public void SortByDesc()
```

Comment:
Single sort + CompareTo (like on lection)
You don't require to use different sorts here

Reply:
fixed